src/DGtal/geometry/doc/moduleHull2D.dox

```diff
@@ -0,0 +1,307 @@
+@image latex ConvexHullCW.png "Clockwise-oriented list of extremal points of the digitization of a disk" width=5cm
+
+
+\subsection subsectmoduleHull2D12 Graham's algorithm
+
+Graham's algorithm is another well-known algorithm to compute the convex hull of 
+a 2D point set of size \f$ n \f$ in \f$ O(n\log{(n)}) \f$ [Graham, 1972 : \cite Graham1972]. 
+The method can be coarsely describe as follows: 
+- choose a pole and sort the points in order of increasing angle about the pole
+- apply a simple Graham's scan on the sorted list of points. 
+
+We have chosen the point of maximal x-coordinate (and maximal y-coordinate) to be the pole. 
+Hence, the algorithm is correct because sorting the points in order of increasing angle about 
+such a pole produces again a WEVP. 
+ 
+@image html GrahamWEVP.png "Sorting step of Graham's algorithm produces a WEVP"
+@image latex GrahamWEVP.png "Sorting step of Graham's algorithm produces a WEVP" width=5cm
+
+The main advantage of Graham's algorithm over Andrew's one is that only one scan is required 
+instead of two to retrieve the whole convex hull. 
+
+You may call the procedure grahamConvexHullAlgorithm as follows:  
+@snippet geometry/tools/exampleConvexHull2D.cpp Hull2D-GrahamAlgo
+
+@warning You must choose a predicate returning 'true' for counter-clockwise oriented 3-point sets, 
+because the sorting step is done with a counter-clockwise orientation.  
+
+\subsection subsectmoduleHull2D13 Melkman's algorithm
+
+The two previous algorithms are off-line and compute the convex hull of any set
+of \f$ n \f$ 2D points in \f$ O(n\log{(n)}) \f$.  
+Melkman's algorithm is on-line and computes the convex hull of a simple polygonal line, 
+ie. without self-intersection, in linear-time (amortized constant time per update). 
+
+Note that Graham scan, used by the two previous algorithms, runs in linear-time 
+on weakly externally visible polygons, which is a subset of simple polygonal lines.
+As a consequence, Graham scan may fail for simple polygonal lines that are not 
+weakly externally visible.    
+
+In exampleConvexHull2D.cpp, you may find a minimal example: 
+
+@image html SimplePolygonalLine.png "A simple polygonal line that is not weakly externally visible"
+@image latex SimplePolygonalLine.png "A simple polygonal line that is not weakly externally visible" width=5cm
+
+On such example, Graham scan fails because the polygonal line is not weakly externally visible. 
+However, Melkman's algorithm produces the right output: 
+
+@image html SimplePolygonalLineMelkman.png "Output for Melkman's algorithm "
+@image latex SimplePolygonalLineMelkman.png "Output for Melkman's algorithm " width=5cm
+
+Melkman's algorithm works fine because it is based on a double ended queue and it  
+assumes that the input points form a simple polygonal line. Due to this assumption, 
+any new point cannot be located in the cone formed by the first and last edge of the 
```

Comment:
does the Melkman's API provide methods to add/remove points and maintain the cvxhull? 
it does not seem so.